site/zenodo_rdm/legacy/requests/record_upgrade.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo legacy record upgrade request."""
+
+from invenio_access.permissions import system_identity
+from invenio_i18n import gettext as _
+from invenio_rdm_records.proxies import current_record_communities_service
+from invenio_records_resources.services.uow import IndexRefreshOp, RecordCommitOp
+from invenio_requests.customizations import CommentEventType, RequestType, actions
+from invenio_requests.proxies import current_events_service, current_requests_service
+
+
+def _remove_permission_flag(record):
+    """Remove can_community_read_files permission flag."""
+    try:
+        record.parent.permission_flags.pop("can_community_read_files")
+        record.parent.commit()
+    except (KeyError, AttributeError):
+        # if field or flag is absent just continue with an action
+        pass
+
+
+def _remove_record_from_communities(record, identity):
+    """Remove record from all the communities."""
+    communities_ids = record.parent.communities.ids
+    communities = []
+    for community_id in communities_ids:
+        communities.append({"id": community_id})
+
+    data = dict(communities=communities)
+    current_record_communities_service.remove(identity, record["id"], data)
+
+
+#
+# Actions
+#
+class SubmitAction(actions.SubmitAction):
+    """Submit action."""
+
+    def execute(self, identity, uow):
+        """Execute the submit action."""
+        record = self.request.topic.resolve()
+        # Set the record's title as the request title.
+        self.request["title"] = record.metadata["title"]
+        super().execute(identity, uow)
+
+
+class AcceptAction(actions.AcceptAction):
+    """Accept action."""
+
+    def execute(self, identity, uow):
+        """Grant access to restricted files of a record for a community."""
+        record = self.request.topic.resolve()
+
+        # remove flag from record parent, permissions logic will do the rest
+        _remove_permission_flag(record)
```

Comment:
I would add an extra comment (event), stating something like:
`{user first name/last name} accepted the request. The community {name} will now have access to the restricted files.` or similar.